python/perf-kernels/hbm-bw-test.py

```diff
@@ -0,0 +1,106 @@
+"""
+Simple test to measure achieved HBM bandwidth.
+This kernel moves N bytes of data from one region in HBM to another, using Triton.
+"""
+
+# %%
+# Compute Kernel
+# --------------
+
+import torch
+
+import triton
+import triton.language as tl
+
+
+@triton.jit
+def copy_kernel(
+    input_ptr,  # *Pointer* to input vector.
+    output_ptr,  # *Pointer* to output vector.
+    n_elements: tl.constexpr,  # Total elements to move.
+    BLOCK_SIZE: tl.constexpr,  # Elements to load / store per iteration
+    vector_size: tl.constexpr, # Size of the entire vector being moved.
+    BOUNDS_CHECK: tl.constexpr, # Whether to use mask for loads.
+
+):
+    pid = tl.program_id(axis=0)
+    
+    lo = pid * n_elements
+    hi = lo + n_elements
+    for idx in range(lo, hi, BLOCK_SIZE):
+        offsets = idx + tl.arange(0, BLOCK_SIZE)
+        # Create a mask to guard memory operations against out-of-bounds accesses.
+        if BOUNDS_CHECK:
+            mask = offsets < vector_size
+        in_vals = tl.load(input_ptr + offsets, mask=mask if BOUNDS_CHECK else None)
+        tl.store(output_ptr + offsets, in_vals, mask=mask if BOUNDS_CHECK else None)
+
+
+# %%
+# Let's also declare a helper function to (1) allocate the `z` tensor
+# and (2) enqueue the above kernel with appropriate grid/block sizes:
+
+
+def copy(x: torch.Tensor, wgs=512):
+    # We need to preallocate the output.
+    output = torch.empty_like(x)
+    assert x.is_cuda
+    vector_size = output.numel()
+    BLOCK_SIZE = 16384
+    grid = (wgs, 1, 1)
+    BOUNDS_CHECK = True
+    # Each WG will move these many elements
+    n_elements = triton.cdiv(vector_size, wgs)
+    copy_kernel[grid](
+        x, output,
+        n_elements, BLOCK_SIZE=BLOCK_SIZE,
+        vector_size=vector_size, BOUNDS_CHECK=BOUNDS_CHECK,
+        num_warps=4
+    )
+    return output
+
+
+torch.manual_seed(0)
+size = 2**30
+x = torch.rand(size, device='cuda')
+output_torch = x
+output_triton = copy(x)
+print(output_torch)
+print(output_triton)
+print(
+    f'The maximum difference between torch and triton is '
+    f'{torch.max(torch.abs(output_torch - output_triton))}'
+)
+
+configs = []
+for wgs in [2 ** i for i in range(0, 12)]:
+    configs.append(triton.testing.Benchmark(
+        x_names=['size'],  # Argument names to use as an x-axis for the plot.
```

Comment:
It looks better if you put 'wgs' in x_names. Here is an example output:
```
size=1073741824:
       wgs       Triton        Torch
0      1.0     7.344461  3314.061557
1      2.0    14.745277  3320.736328
2      4.0    28.924817  3322.230003
3      8.0    57.588137  3313.546153
4     16.0   114.162221  3305.674749
5     32.0   224.866428  3299.158510
6     64.0   437.931297  3303.634809
7    128.0   843.094580  3306.158885
8    256.0  1598.725318  3301.853968
9    512.0  2786.862419  3298.091823
10  1024.0  2785.014515  3302.973052
11  2048.0  2816.407047  3295.605970
```

Reply:
Thanks for the suggestion - it looks better that way indeed. Done.